docs/proposals/resource-catalog.md

```diff
@@ -0,0 +1,222 @@
+
+
+We can reshape the catalog resource to more easily express them, secure them, and claim them.
+
+# Goals
+ 1.  cluster-admin should control which catalog exist.
+ 2.  cluster-admin should be able/disable per-catalog exposure on a per-project granularity.
+ 3.  project-admin should be able to delegate per-catalog exposure.
+ 4.  project-editor should be able to claim an entry from the catalog.
+ 5.  ResourceCatalogClaims should be sync-able, but not require syncing.
+ 6.  project-admin should be able to control the shape of claim realization and prevent unwanted injection.
+ 7.  ResourceCatalogEntries should not be restricted to specific claim realization mechanism.
+
+To achieve the ACL related goals, we can create namespaced resources that are selected by label at a cluster level.
+
+
+# API Objects
+
+## ResourceCatalog
+ResourceCatalog is a cluster-scoped resource.  It has a cluster-admin controlled spec that is used to drive a controller.
+The spec includes a label selector that is used in a list/watch to filter all ResourceCatalogEntries in the cluster.
+The controller will find matching entries and copy a reference into the ResourceCatalog.Status field.  The controller
```

Comment:
Can you give an example?
On Mon, Apr 25, 2016 at 12:34 PM David Eads <notifications@github.com>
wrote:

> In docs/proposals/resource-catalog.md
> <https://github.com/ncdc/kubernetes/pull/8#discussion_r60944675>:
>
> > + 2.  cluster-admin should be able/disable per-catalog exposure on a per-project granularity.
> > + 3.  project-admin should be able to delegate per-catalog exposure.
> > + 4.  project-editor should be able to claim an entry from the catalog.
> > + 5.  ResourceCatalogClaims should be sync-able, but not require syncing.
> > + 6.  project-admin should be able to control the shape of claim realization and prevent unwanted injection.
> > + 7.  ResourceCatalogEntries should not be restricted to specific claim realization mechanism.
> > +
> > +To achieve the ACL related goals, we can create namespaced resources that are selected by label at a cluster level.
> > +
> > +
> > +# API Objects
> > +
> > +## ResourceCatalog
> > +ResourceCatalog is a cluster-scoped resource.  It has a cluster-admin controlled spec that is used to drive a controller.
> > +The spec includes a label selector that is used in a list/watch to filter all ResourceCatalogEntries in the cluster.
> > +The controller will find matching entries and copy a reference into the ResourceCatalog.Status field.  The controller
>
> Yeah that's why I'd prefer actual entries
>
> This design has actual entries, but as I noted you get stuck with a
> resourceversion nightmare if you don't pull them in.
>
> You didn't address the problem in your design. How would you resolve the
> problem of non-coherent resource versions from multiple non-uniform
> objects? We're facing the same problem downstream and I couldn't come up
> with anything better than a partial ordering.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/ncdc/kubernetes/pull/8/files/a2acd970c6316ce196c7bb3f24fb57e1bbef647e#r60944675>
>